src/link/graphql_definition.rs

```diff
@@ -70,3 +80,27 @@ impl Display for BooleanOrVariable {
         }
     }
 }
+
+impl BooleanOrVariable {
+    pub(crate) fn to_ast_value(&self) -> Value {
+        match self {
+            BooleanOrVariable::Boolean(b) => Value::Boolean(*b),
+            BooleanOrVariable::Variable(name) => Value::Variable(name.clone()),
+        }
+    }
+}
+
+impl From<BooleanOrVariable> for Value {
+    fn from(b: BooleanOrVariable) -> Self {
+        match b {
+            BooleanOrVariable::Boolean(b) => Value::Boolean(b),
+            BooleanOrVariable::Variable(name) => Value::Variable(name),
+        }
+    }
+}
+
+impl From<BooleanOrVariable> for Node<Value> {
+    fn from(b: BooleanOrVariable) -> Self {
+        b.into()
```

Comment:
I *think* this impls infinitely calls itself via the standard library’s blanket `impl<T, U> Into<U> for T where U: From<T>`

Instead, consider converting to `Value` then converting to `Node<Value>`:
```suggestion
impl From<BooleanOrVariable> for Node<Value> {
    fn from(b: BooleanOrVariable) -> Self {
        Node::new(b.into())
```